ts/utils/profile.ts

```diff
@@ -0,0 +1,85 @@
+import { format, format as dateFnsFormat } from "date-fns";
+import * as pot from "italia-ts-commons/lib/pot";
+import { FiscalCode } from "../../definitions/backend/FiscalCode";
+import { MunicipalityState } from "../store/reducers/content";
+
+type GenderType = "M" | "F" | undefined;
+
+/**
+ * Generic utilities for profile
+ */
+type FiscalCodeDerivedData = Readonly<{
+  gender?: GenderType;
+  birthDate?: ReturnType<typeof dateFnsFormat>;
+  denominazione: string;
+  siglaProvincia: string;
+}>;
+
+// Generate object including data expressed into the given fiscal code
+export function extractFiscalCodeData(
+  fiscalCode: FiscalCode,
+  municipality: MunicipalityState
+): FiscalCodeDerivedData {
+  const siglaProvincia = pot.isSome(municipality.data)
+    ? municipality.data.value.siglaProvincia
+    : "";
+  const denominazione = pot.isSome(municipality.data)
+    ? municipality.data.value.denominazioneInItaliano
+    : "";
+  // tslint:disable-next-line: no-let
+  let gender: GenderType;
+  // tslint:disable-next-line: no-let
+  let birthDate;
+  if (!RegExp("^[0-9]+$").test(fiscalCode.substring(9, 11))) {
+    return {
+      gender,
+      birthDate,
+      siglaProvincia,
+      denominazione
+    };
+  }
+
+  const tempDay = parseInt(fiscalCode.substring(9, 11), 10);
+  gender = tempDay - 40 > 0 ? "F" : "M";
+  const months: { [k: string]: number } = {
```

Comment:
You should move this mapping outside this function as a const.
So it will be create only once

Reply:
updated within d0ea70c2f64bc9fc4a091b528d998af71f43ee3c